dask/array/core.py

```diff
@@ -1156,6 +1156,24 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
             if result is not None:
                 self = result
 
+        try:
+            layer = self.dask.layers[name]
+        except (AttributeError, KeyError):
+            # self is no longer an Array after applying the plugins, OR
+            # a plugin replaced the HighLevelGraph with a plain dict, OR
+            # name is not the top layer's name (this can happen after the layer is
+            # manipulated, to avoid a collision)
+            pass
+        else:
+            layer.collection_annotations.update(
+                {
+                    "type": type(self),
+                    "chunk_type": type(meta),
```

Comment:
Nitpick: What you have is totally fine and `type(meta)` should be the same as `type(self._meta)`. Though this may change in the future if we add extra code between where `meta` and `self._meta` are defined above. Unless there's a reason not to (which I may be missing) then I'd prefer to use `self._meta` here to keep things more robust to future changes. 

```suggestion
                    "chunk_type": type(self._meta),
```